### PR TITLE
Test fixes for C++ client

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

File: tests/test_update.py [hunk 1]
```diff
@@ -200,12 +214,12 @@ def test_too_big_image_update(self, bitbake_variables, connection):
...
             )
             output = connection.run(
-                "mender install /var/tmp/image-too-big.mender ; echo 'ret_code=$?'"
+                f"{mender_update_binary} install /var/tmp/image-too-big.mender ; echo ret_code=$?"
             )
 
             assert any(
                 [
-                    "no space left on device" in out
+                    "no space left on device" in out or "No space left on device" in out
```
Comment (oleorhagen) [inline tests/test_update.py:223]: S: `no space left on device in out.lower()`

File: tests/test_update.py [hunk 2]
```diff
@@ -611,6 +612,13 @@ def test_signed_updates(
         """Test various combinations of signed and unsigned, present and non-
         present verification keys."""
 
+        if sig_case.artifact_version == 2 and version_is_minimum(
+            bitbake_variables, "mender", "4.0.0"
+        ):
+            pytest.skip(
```
Comment (oleorhagen) [inline tests/test_update.py:624]: Q: Why not skipif?

File: tests/test_update.py [hunk 3]
```diff
@@ -612,6 +612,12 @@ def test_signed_updates(
         """Test various combinations of signed and unsigned, present and non-
         present verification keys."""
 
+        if (
+            sig_case.label
+            == "EC, Correctly signed, but header does not match checksum, key present"
+        ):
+            pytest.skip("MEN-6845")
```
Comment (oleorhagen) [inline tests/test_update.py:619]: `skipif`?

**Review by oleorhagen — APPROVED** [review]
Looks good I think :smile:

File: tests/test_update.py (hunk 2 above)
Comment (kacf) [reply tests/test_update.py:624]: Skipif doesn't support using fixture variables in the expression, and I need `bitbake_variables`.

File: tests/test_update.py (hunk 3 above)
Comment (kacf) [reply tests/test_update.py:619]: I think I'll leave it simply because it's similar to the other one. It will go away pretty soon anyway.

**Review by lluiscampos — APPROVED** [review]

File: tests/test_update.py [hunk 4]
```diff
@@ -714,9 +732,14 @@ def test_signed_updates(self, sig_case, bitbake_variables, connection):
...
 
                     result = connection.run(
-                        "mender install /data/image.mender", warn=True
+                        f"{mender_update_binary} install /data/image.mender", warn=True
                     )
 
+                    if result.return_code == 0:
+                        # Just reset database for next update. The content is still there, IOW it
+                        # doesn't wipe it, and we can still test it.
+                        connection.run(f"{mender_update_binary} rollback")
```
[the remark this answers is not in the document]
Comment (kacf) [reply tests/test_update.py:755]: The standalone client automatically rolls back failed updates.

**Pull request merged**